flocker/node/agents/test/test_blockdevice.py

```diff
@@ -982,12 +983,13 @@ def test_no_devices_one_dataset(self):
             nodes={
                 Node(
                     uuid=uuid,
-                    hostname=node,
                     manifestations={dataset_id: manifestation},
                 )
             }
         )
-        state = DeploymentState(nodes=[])
+        state = DeploymentState(nodes=[NodeState(
+            uuid=uuid, hostname=node, applications=[], manifestations={},
+            devices={}, paths={})])
```

Comment:
I think I'll just do the invariant in this branch, since it'll reduce the chances of this branch introducing subtle bugs.